app/graphql/resolvers/utils.ts

```diff
@@ -0,0 +1,20 @@
+export type DataSourceType = "sparql" | "sql";
+
+export type DataSource = {
+  type: DataSourceType;
+  url: string;
+};
+
+export const retrieveEndpointFromLocalStorage = () => {};
+
+export const convertEndpointToSource = (endpoint: string): DataSource => {
+  const [type, url] = endpoint.split("+") as [DataSourceType, string];
+
+  return { type, url };
+};
+
+export const convertSourceToEndpoint = (source: DataSource): string => {
+  const { type, url } = source;
+
+  return `${type}+${url}`;
+};
```

Comment:
Nit on the naming, having stringifySource and parseSource  (or stringifyEndpoint parseEndpoint) would be a bit clearer and would remove the need to have both source and endpoint.

Reply:
Good point, will adjust the logic 👌